src/autopas/containers/linkedCells/traversals/LCC08CellHandlerUtility.cpp

```diff
@@ -0,0 +1,189 @@
+/**
+ * @file LCC08CellHandlerUtility.cpp
+ * @author J. Schuhmacher
+ * @date 11.07.2024
+ */
+
+#include "LCC08CellHandlerUtility.h"
+
+namespace autopas::internal {
+
+constexpr int calculateCell2Index(int overlap1, const C08CellDirection &direction, int z) {
+  switch (direction) {
+    case C08CellDirection::FRONT_LEFT:
+      return z;
+    case C08CellDirection::BACK_LEFT:
+      return overlap1 * (overlap1 - 1) + z;
+    case C08CellDirection::FRONT_RIGHT:
+      return overlap1 * overlap1 * (overlap1 - 1) + z;
+    case C08CellDirection::BACK_RIGHT:
+      return overlap1 * overlap1 * overlap1 - overlap1 + z;
+  }
+  return 0;
+}
+
+constexpr std::pair<int, int> toDistVectorMultiplier(const C08CellDirection &direction) {
+  switch (direction) {
+    case C08CellDirection::FRONT_LEFT:
+      return std::make_pair(0, 0);
+    case C08CellDirection::BACK_LEFT:
+      return std::make_pair(0, 1);
+    case C08CellDirection::FRONT_RIGHT:
+      return std::make_pair(1, 0);
+    case C08CellDirection::BACK_RIGHT:
+      return std::make_pair(1, 1);
+  }
+  return std::make_pair(0, 0);
+}
+
+constexpr bool includeC08Case(const C08CellDirection &direction, const std::array<int, 3> &overlap, int x, int y,
+                              int z) {
+  switch (direction) {
+    case C08CellDirection::BACK_LEFT:
+      return y != overlap[1] and z != 0;
+    case C08CellDirection::FRONT_RIGHT:
+      return x != overlap[0] and (y != 0 or z != 0);
+    case C08CellDirection::BACK_RIGHT:
+      return y != overlap[1] and x != overlap[0] and z != 0;
+    case C08CellDirection::FRONT_LEFT:
+    default:
```

Comment:
```suggestion
    case C08CellDirection::FRONT_LEFT:
      [[fallthrough]];
    default:
```
Otherwise some compilers might throw warnings.

Alternatively don't have a default case and a dedicated return after the switch as in the function above. Either way I think both functions should use the same pattern.
Advantage of not having a `default` case: If a new enum value is introduced you get a compiler warning for every switch that doesn't cover all cases.